io/plugins/eu.esdihumboldt.hale.io.shp.test/src/eu/esdihumboldt/hale/io/shp/ShapefileInstanceWriterTest.groovy

```diff
@@ -1443,4 +1456,24 @@ class ShapefileInstanceWriterTest {
 			assertEquals(2, num)
 		}
 	}
+
+	@Test
+	void testCPGFileCreation() {
+		def tmpDir = Files.createTempDirectory("CPGFileTest")
+		def tmpFile = Files.createTempFile(tmpDir, 'new', '.cpg')
+
+		try {
+			ShapefileInstanceWriter writer = new ShapefileInstanceWriter();
+			writer.writeCodePageFile(tmpFile.toFile().getAbsolutePath());
+			assertTrue(tmpFile.toFile().exists());
+
+			String expected_value = writer.getCharset().toString();
+			BufferedReader bufferedReader = new BufferedReader(new FileReader(tmpFile.toFile().getAbsolutePath()));
```

Comment:
Just an FYI - since we use Groovy here, there is a simple way to get the text content of a file:
https://docs.groovy-lang.org/latest/html/groovy-jdk/java/io/File.html#getText()